Automattic/jetpack
# Boost: Add transitions for critical css advanced recommendations

> **Warning**
> Needs to be merged after https://github.com/Automattic/jetpack/pull/35109, as it depends on changes introduced by it.





Refer to https://github.com/Automattic/jetpack/issues/35015

I'm not entirely happy with the jumping, but I couldn't find a way around it.

## Proposed changes:

* Add transitions to Critical CSS Advanced section.
  * Heading text;
  * "Show hidden recommendations" link;
  * Ignoring a recommendation (collapse and expand).

<details><summary>Preview</summary>
<img src="https://github.com/Automattic/jetpack/assets/11799079/9e33238e-281f-4d92-a2eb-a8135aa4e9c3" />
</details>


### Other information:


## Jetpack product discussion



https://github.com/Automattic/jetpack/issues/35015

## Does this pull request change what data or activity we track or use?



no

## Testing instructions:






* Setup Boost (free);
* Force Critical CSS generation to show the advanced recommendations page by adding this snippet to your site and regenerating critical CSS:

```php
<?php

add_action( 'template_redirect', function () {
	if ( ! is_front_page() ) {
		noexistentfunction();
	}
} );
```

* Interact with the transitions and the "Show recommendations" link and observe the transitions.
